client/src/main/java/org/apache/celeborn/client/write/DataPushQueue.java

```diff
@@ -99,15 +99,19 @@ public ArrayList<PushTask> takePushTasks() throws IOException {
             client.getPartitionLocation(appId, shuffleId, numMappers, numPartitions);
         if (partitionLocationMap != null) {
           PartitionLocation loc = partitionLocationMap.get(partitionId);
-          Integer oldCapacity = workerCapacity.get(loc.hostAndPushPort());
-          if (oldCapacity == null) {
-            oldCapacity = maxInFlight - pushState.inflightPushes(loc.hostAndPushPort());
-            workerCapacity.put(loc.hostAndPushPort(), oldCapacity);
-          }
-          if (oldCapacity > 0) {
-            iterator.remove();
+          if (loc != null) {
```

Comment:
IMO it shouldn't happen that partitionLocationMap is not null but loc is null. RegisterShuffle is guaranteed to be success or fail, no partial success. Maybe we should check why it happens in your environment.

Reply:
Let me check.

Reply:
I have checked, the logic seems the same.

Reply:
> 

After discuss with @AngersZhuuuu , I recalled that in speculation scenarios, it can happen that the location is null. See description in https://github.com/apache/incubator-celeborn/pull/1468 . So I think this PR is good.